### DISCUSSION TITLE
How to instruct GitExtensions to use core.useBuiltinFSMonitor=true

### DISCUSSION THREAD

**Discussion by fabvit** [gitextensions/gitextensions#9023] [Q&A] [answered]
Just installed GitExtensions v3.5-RC1 along Git for Windows v2.31.0.
I am wondering if there is some config setting in GitExtensions to instruct it to use the new Git for windows experimental built-in file-system monitor? I've already added core.useBuiltinFSMonitor=true to my global .gitconfig.

**Comment by pmiossec** [comment, marked as answer]
No, no support added at the moment. 

But I don't think that it is a good idea to do something now when I see the warning :

>NOTE! This command is still only an experiment, subject >to change dramatically
>(or even to be abandoned).

But what do you think GitExtensions should do to support it? How this new feature should be used from a GUI? What is the added value?

The new feature PR: https://github.com/git-for-windows/git/pull/3082

**Reply by pmiossec** [reply]
From what I see, GUIs has nothing to change. It's between the deamon and the `git status` command.

So except if I'm wrong no support has to be added in GitExtensions to support it...

**Reply by fabvit** [reply]
> 
> 
> From what I see, GUIs has nothing to change. It's between the deamon and the `git status` command.
> 
> So except if I'm wrong no support has to be added in GitExtensions to support it...

Thank you, this answered my question.
I will keep experimenting if there is some visible benefits leaving this setting enabled.

**Comment by gerhardol** [comment]
If enabled from the commandline (as many other Git settings), this should be available.
The fsmonitor is used by Git internally to speed up commands like `git-status`, so I see no direct support needed in GE.
(Of course, there could be a GUI checkbox in Settings - Git - Advanced, but you only set this once...)

Has anyone seen good improvements with this or any existing external monitor like `watchman`?

It would be nice if the built in monitor could replace the FileSystemWatcher used to trigger git-status at changes, but there does not seem to be an interface for that right now.

**Reply by fabvit** [reply]
>It would be nice if the built in monitor could replace the FileSystemWatcher used to trigger git-status at changes, but there does >not seem to be an interface for that right now.

Yes I agree on this.
